### PR TITLE
Added docker-machine support

### PR DESCRIPTION
**Pull request by sq4ind**
Added docker-machine support. Hopefully I haven't messed up again with commits and with all the other things (sorry for so many pull requests).

molecule can use the env variables to spin up docker containers on docker-machine environment. It is completely transparent for the molecule (using variables from env or setting it up in molecule.yml).

#### PR Type

- Feature Pull Request

### CODE REVIEW

File: molecule/provisioner/ansible/playbooks/docker/create.yml [hunk 1]
```diff
@@ -12,6 +12,10 @@
         email: "{{ item.registry.credentials.email | default(omit) }}"
         registry: "{{ item.registry.url }}"
         docker_host: "{{ item.docker_host | default(lookup('env', 'DOCKER_HOST') or 'unix://var/run/docker.sock') }}"
+        cacert_path: "{{ item.cacert_path | default((lookup('env', 'DOCKER_CERT_PATH')+'/ca.pem') if lookup('env', 'DOCKER_CERT_PATH') else omit) }}"
```
Comment (webknjaz) [inline molecule/provisioner/ansible/playbooks/docker/create.yml]: Please wrap operators with spaces:
```suggestion
        cacert_path: "{{ item.cacert_path | default((lookup('env', 'DOCKER_CERT_PATH') + '/ca.pem') if lookup('env', 'DOCKER_CERT_PATH') else omit) }}"
```

Comment (sq4ind) [reply molecule/provisioner/ansible/playbooks/docker/create.yml]: @webknjaz All should be done.

File: molecule/driver/docker.py [hunk 2]
```diff
@@ -86,6 +86,10 @@ class Docker(base.Base):
             network_mode: host
             purge_networks: true
             docker_host: tcp://localhost:12376
+            cacert_path: /foo/bar/ca.pem
+            cert_path: /foo/bar/cert.pem
+            key_path: /foo/bar/key.pem
+            tls_verify: 1
```
Comment (decentral1se) [inline molecule/driver/docker.py]: As far as I know, we're using `True` more regularly here ...

File: molecule/model/schema_v2.py [hunk 3]
```diff
@@ -737,6 +737,22 @@ def pre_validate_base_schema(env, keep_string):
...
+                },
+                'cert_path': {
+                    'type': 'string',
+                },
+                'key_path': {
+                    'type': 'string',
+                },
+                'tls_verify': {
+                    'type': 'integer',
+                    'regex': '^[0-1]$',
```
Comment (decentral1se) [inline molecule/model/schema_v2.py]: This should be a boolean or?

File: molecule/provisioner/ansible/playbooks/docker/create.yml [hunk 4]
```diff
@@ -30,6 +34,11 @@
       docker_image_facts:
         name: "molecule_local/{{ item.item.name }}"
         docker_host: "{{ item.item.docker_host | default(lookup('env', 'DOCKER_HOST') or 'unix://var/run/docker.sock') }}"
+        cacert_path: "{{ item.cacert_path | default((lookup('env', 'DOCKER_CERT_PATH') + '/ca.pem') if lookup('env', 'DOCKER_CERT_PATH') else omit) }}"
+        cert_path: "{{ item.cert_path | default((lookup('env', 'DOCKER_CERT_PATH') + '/cert.pem')  if lookup('env', 'DOCKER_CERT_PATH') else omit) }}"
+        key_path: "{{ item.key_path | default((lookup('env', 'DOCKER_CERT_PATH') + '/key.pem') if lookup('env', 'DOCKER_CERT_PATH') else omit) }}"
+        tls_verify: "{{ item.tls_verify | default(lookup('env', 'DOCKER_TLS_VERIFY')) or '0' }}"
+        debug: true
```
Comment (decentral1se) [inline molecule/provisioner/ansible/playbooks/docker/create.yml]: Hmmm, why is this here? We typically rely on Molecule log/no_log configuration for this.

Comment (sq4ind) [reply molecule/provisioner/ansible/playbooks/docker/create.yml]: Sorry, I was debugging it and I've just omitted this one. Removed.

File: molecule/model/schema_v2.py (hunk 3 above)
Comment (sq4ind) [reply molecule/model/schema_v2.py]: Fixed.

File: molecule/driver/docker.py (hunk 2 above)
Comment (sq4ind) [reply molecule/driver/docker.py]: Yes that `True` , fixed this one as well

File: test/unit/model/v2/test_platforms_section.py [hunk 5]
```diff
@@ -262,6 +277,11 @@ def test_platforms_docker_has_errors(_config):
                 'env': ['must be of dict type'],
                 'restart_policy': ['must be of string type'],
                 'restart_retries': ['must be of integer type'],
+                'docker_host': ['must be of string type'],
+                'cacert_path': ['must be of string type'],
+                'cert_path': ['must be of string type'],
+                'key_path': ['must be of string type'],
+                'tls_verify': ['must be of integer type'],
```
Comment (webknjaz) [inline test/unit/model/v2/test_platforms_section.py]: Integer? I thought boolean...

File: test/unit/model/v2/test_platforms_section.py [hunk 6]
```diff
@@ -200,6 +210,11 @@ def _model_platforms_docker_errors_section_data():
             ],
             'network_mode': int(),
             'purge_networks': int(),
+            'docker_host': str(),
+            'cacert_path': str(),
+            'cert_path': str(),
+            'key_path': str(),
+            'tls_verify': int(),
```
Comment (webknjaz) [inline test/unit/model/v2/test_platforms_section.py]: Maybe
```suggestion
            'tls_verify': bool(),
```
?

File: test/unit/model/v2/test_platforms_section.py [hunk 7]
```diff
@@ -104,7 +104,17 @@ def _model_platforms_docker_section_data():
...
+            'docker_host':
+            'tcp://localhost:12376',
+            'cacert_path':
+            '/foo/bar/ca.pem',
+            'cert_path':
+            '/foo/bar/cert.pem',
+            'key_path':
+            '/foo/bar/key.pem',
+            'tls_verify':
+            '1'
```
Comment (webknjaz) [inline test/unit/model/v2/test_platforms_section.py]:
```suggestion
            True,
```

File: test/unit/model/v2/test_platforms_section.py [hunk 8]
```diff
@@ -104,7 +104,17 @@ def _model_platforms_docker_section_data():
...
+            'docker_host':
+            'tcp://localhost:12376',
+            'cacert_path':
+            '/foo/bar/ca.pem',
+            'cert_path':
+            '/foo/bar/cert.pem',
+            'key_path':
+            '/foo/bar/key.pem',
+            'tls_verify':
+            'true'
```
Comment (webknjaz) [inline test/unit/model/v2/test_platforms_section.py]: It should be boolean `True`, not a string `'true'`
```suggestion
            True
```
**Please just click a button in this suggestion, this will apply the change right from the GitHub UI**

**Comment by sq4ind** [comment]
Hey guys something else should be done to this PR to get it merged? (I know that CI is failing but have not idea why). Thanks

**Comment by decentral1se** [comment]
Looks fine last I checked but we do need to fix our CI now (unrelated)  ... thanks for patience.

**Comment by sq4ind** [comment]
> Looks fine last I checked but we do need to fix our CI now (unrelated) ... thanks for patience.

No worries. Let me know if I can help you somehow.

**Comment by webknjaz** [comment]
@sq4ind I've restarted the build. In future be aware that you can trigger CI by just closing and reopening the PR :)

**Comment by webknjaz** [comment]
Hey @sq4ind, could you please fix the failing unit test?

```console
test/unit/model/v2/test_platforms_section.py::test_platforms_docker_has_errors[_model_platforms_docker_error...] FAILED [ 69%]
```
https://travis-ci.com/ansible/molecule/jobs/200892320#L964

**Comment by sq4ind** [comment]
> Hey @sq4ind, could you please fix the failing unit test?
> 
> ```
> test/unit/model/v2/test_platforms_section.py::test_platforms_docker_has_errors[_model_platforms_docker_error...] FAILED [ 69%]
> ```
> 
> https://travis-ci.com/ansible/molecule/jobs/200892320#L964

I believe unit test are now fixed. Sorry for taking so long time to fix them.

**Comment by sq4ind** [comment]
@webknjaz  I think the build system went wrong. Am correct?

**Comment by sq4ind** [comment]
Hi, can this be merged now ?

Thanks.
